src/baseInterceptor.js

```diff
@@ -1,252 +1,232 @@
-function baseInterceptor() {
-  this.prevTotalCount = 0,
-  this.totalCount = 0,
-  this.currentColor = 'white',
-  this.bgColor = 'white',
-  this.objectArea = 0,
-  this.coordinates = [],
-  this.objectDescription = '',
-  this.canvasDetails = {
-    width: 0,
-    height: 0
-  },
-  this.setupObject = {
-    objectArray: [],
-    objectCount: 0,
-    objectTypeCount: {}
-  },
-  this.drawObject = {
-    objectArray: [],
-    objectCount: 0,
-    objectTypeCount: {}
-  },
-  this.isCleared = false;
+class baseInterceptor {
+    constructor() {
+        this.prevTotalCount = 0,
+            this.totalCount = 0,
+            this.currentColor = 'white',
+            this.bgColor = 'white',
+            this.objectArea = 0,
+            this.coordinates = [],
+            this.objectDescription = '',
+            this.canvasDetails = {
+                width: 0,
+                height: 0
+            },
+            this.setupObject = {
+                objectArray: [],
+                objectCount: 0,
+                objectTypeCount: {}
+            },
+            this.drawObject = {
+                objectArray: [],
+                objectCount: 0,
+                objectTypeCount: {}
+            },
+            this.isCleared = false;
+    }
+    getColorName(arguments) {
+        if (arguments.length == 4) {
```

Comment:
@spiray can you change the variable 'arguments' to 'colorValues'? It should be a quick fix to the error we are getting. Make the change throughout the whole file 😄

Reply:
This is complete